# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/FileManagerBase.h
# Commit: bb094a6084b3
# Full Hash: bb094a6084b3b127c090827ee8ede4b06cdbdea6
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-03-19 17:04:52
# Regressor Bug: 1623278
# File Overlap Count: 1
# Description:
#   Bug 1623278 - Use SafeRefPtr for FileManager. r=dom-workers-and-storage-reviewers,asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D65549
# ==============================================================================

diff -r 7fdf38a403bb -r bb094a6084b3 dom/indexedDB/FileManagerBase.h
--- a/dom/indexedDB/FileManagerBase.h	Thu Mar 19 09:32:02 2020 +0000
+++ b/dom/indexedDB/FileManagerBase.h	Thu Mar 19 09:32:53 2020 +0000
@@ -60,8 +60,10 @@
 
       const int64_t id = ++mLastFileId;
 
-      fileInfo =
-          new FileInfo(FileManagerGuard{}, static_cast<FileManager*>(this), id);
+      fileInfo = new FileInfo(FileManagerGuard{},
+                              SafeRefPtr{static_cast<FileManager*>(this),
+                                         AcquireStrongRefFromRawPtr{}},
+                              id);
 
       mFileInfos.Put(id, fileInfo);
     }